File: wmc_event-planner/fix_storage.py
#!/usr/bin/env python3
import re

with open('projektplanung.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Replace JSON.parse(localStorage.getItem('projects')) with getAllProjectsSync()
content = re.sub(
    r'JSON\.parse\(localStorage\.getItem\("projects"\) \|\| "{}"\)',
    'getAllProjectsSync() || {}',
    content
)

# Replace localStorage.setItem('projects', JSON.stringify(projects)) with saveProjectsData(projects)
content = re.sub(
    r'localStorage\.setItem\("projects", JSON\.stringify\(projects\)\)',
    'saveProjectsData(projects)',
    content
)

with open('projektplanung.html', 'w', encoding='utf-8') as f:
    f.write(content)

print('✓ All storage calls replaced in projektplanung.html')

File: fix_storage.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'projektplanung.html');
let content = fs.readFileSync(filePath, 'utf-8');

// Replace JSON.parse(localStorage.getItem('projects')) with getAllProjectsSync()
content = content.replace(
    /JSON\.parse\(localStorage\.getItem\("projects"\) \|\| "{}"\)/g,
    'getAllProjectsSync() || {}'
);

// Replace localStorage.setItem('projects', JSON.stringify(projects)) with saveProjectsData(projects)
content = content.replace(
    /localStorage\.setItem\("projects", JSON\.stringify\(projects\)\)/g,
    'saveProjectsData(projects)'
);

fs.writeFileSync(filePath, content, 'utf-8');
console.log('✓ All storage calls replaced in projektplanung.html');
